### ISSUE TITLE
"NoSpaceBeforeCloseParenthesis" not fired for function declarations in interfaces.

### ISSUE THREAD

**Issue by irshadahmad21** [WordPress/WordPress-Coding-Standards#2204]
The bug, first introduced in #617 and then in #1262 disables the "`NoSpaceBeforeCloseParenthesis`" error for function declarations in interfaces.

## Bug Description
Not having a space before closing parenthesis shows an error for class methods and functions, but it does not do so for the function declarations in interfaces.

## Minimal Code Snippet

The issue happens when running this command:
```bash
phpcs ...
```

... over a file containing this code:
```php
interface SomeInterface {

	/**
	 * Does something
	 */
	public function some_method( $arg1);
	// See no space before )         ^^
}
```

There is no warning about no space before closing parenthesis, while I expect the error to be something like this:

```
ERROR | [x] Expected 1 space before closing parenthesis; 0 found
```

## Custom ruleset

```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="WordPress-Extra" />
</ruleset>
```

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.4
| PHP_CodeSniffer version | 3.7.1
| WPCS version            | 2.3.0
| WPCS install type       | Composer 
| IDE (if relevant)       | VS Code 1.74.3

## Tested Against `develop` branch?
- [x] I have verified the issue still exists in the `develop` branch of WPCS.

**Comment by dingo-d** [comment]
I could reproduce this on the `develop` branch, I think this is a bug. Thanks for reporting it 👍🏼

**Comment by jrfnl** [comment]
Related to/duplicate of #1101. Function declaration statements are currently checked by a sniff which is not suitable for this and which needs to be replaced.

Note: the two PRs referenced by the OP are not directly related to the actual issue.

**Comment by dingo-d** [comment]
Ok, closing as a duplicate of #1101
